zerver/views/streams.py

```diff
@@ -943,43 +936,70 @@ def delete_in_topic(
     return json_success(request, data={"complete": True})
 
 
-@has_request_variables
+@typed_endpoint
 def json_get_stream_id(
-    request: HttpRequest, user_profile: UserProfile, stream_name: str = REQ("stream")
+    request: HttpRequest,
+    user_profile: UserProfile,
+    *,
+    stream_name: Annotated[str, ApiParamConfig("stream")],
 ) -> HttpResponse:
     (stream, sub) = access_stream_by_name(user_profile, stream_name)
     return json_success(request, data={"stream_id": stream.id})
 
 
-@has_request_variables
+class SubscriptionPropertyChangeRequest(BaseModel):
+    stream_id: int
+    property: str
+    value: bool | str
+
+    @model_validator(mode="after")
+    def validate_terms(self) -> "SubscriptionPropertyChangeRequest":
+        boolean_properties = {
+            "in_home_view",
+            "is_muted",
+            "desktop_notifications",
+            "audible_notifications",
+            "push_notifications",
+            "email_notifications",
+            "pin_to_top",
+            "wildcard_mentions_notify",
+        }
+
+        if self.property == "color":
+            self.value = check_color("color", self.value)
+        elif self.property in boolean_properties:
+            if not isinstance(self.value, bool):
+                raise JsonableError(_("{property} is not a boolean").format(property=self.property))
+        else:
+            raise JsonableError(
+                _("Unknown subscription property: {property}").format(property=self.property)
+            )
+        return self
```

Comment:
I refactored this, and placed it above the function using it, just for readability. Logic should be equivalent.

``` diff
diff --git a/zerver/views/streams.py b/zerver/views/streams.py
index 25a76a433e..3986ad51fe 100644
--- a/zerver/views/streams.py
+++ b/zerver/views/streams.py
@@ -962,14 +962,14 @@ def update_subscriptions_property(
     )
 
 
-class SubscriptionData(BaseModel):
+class SubscriptionPropertyChangeRequest(BaseModel):
     stream_id: int
     property: str
     value: bool | str
 
     @model_validator(mode="after")
     def validate_terms(self) -> "SubscriptionData":
-        property_converters = {
+        boolean_properties = {
             "color",
             "in_home_view",
             "is_muted",
@@ -981,15 +981,15 @@ class SubscriptionData(BaseModel):
             "wildcard_mentions_notify",
         }
 
-        if self.property not in property_converters:
-            raise JsonableError(
-                _("Unknown subscription property: {property}").format(property=self.property)
-            )
         if self.property == "color":
             self.value = check_color("color", self.value)
-        else:
+        elif self.property in boolean_properties:
             if not isinstance(self.value, bool):
                 raise JsonableError(_("{property} is not a boolean").format(property=self.property))
+        else:
+            raise JsonableError(
+                _("Unknown subscription property: {property}").format(property=self.property)
+            )
         return self
 
 
@@ -998,7 +998,7 @@ def update_subscription_properties_backend(
     request: HttpRequest,
     user_profile: UserProfile,
     *,
-    subscription_data: Json[list[SubscriptionData]],
+    subscription_data: Json[list[SubscriptionPropertyChangeRequest]],
 ) -> HttpResponse:
     """
     This is the entry point to changing subscription properties. This
```